src/coreclr/vm/precode_portable.hpp

```diff
@@ -31,6 +34,32 @@ class PortableEntryPoint final
 
 public:
     void Init(MethodDesc* pMD);
+
+    // Query methods for entry point state.
+    bool IsValid() const;
+
+    bool IsByteCodeCompiled() const
+    {
+        LIMITED_METHOD_CONTRACT;
+        _ASSERTE(IsValid());
+        return _pInterpreterData != nullptr;
+    }
+
+    bool HasNativeCode() const
+    {
+        LIMITED_METHOD_CONTRACT;
+        _ASSERTE(IsValid());
+        return _pActualCode != nullptr;
+    }
+
+    bool IsReadyForR2R() const
```

Comment:
A nit - would it make sense naming this `IsReadyForAOT` or `IsReadyForAOTCall` instead?